packages/crypto/benches/main.rs

```diff
@@ -157,12 +157,14 @@ where
             .map(|(secret_key, message)| *message * secret_key)
             .collect();
 
-        for i in 1..=two_pow_max {
-            let num_points = 2_usize.pow(i);
-            let messages = &messages[..num_points];
-            let keys = &public_keys[..num_points];
+        for i in 0..=two_pow_max {
+            let n = 2_usize.pow(i); // the number of pairings on the left hand side
+            let k = n + 1; // the number of pairings in total
+            let messages: &[ark_ec::short_weierstrass::Affine<ark_bls12_381::g2::Config>] =
```

Comment:
Let's keep type inference working for us here. 😉
This happens when double-clicking on the type in VS Code.

Reply:
Oh, interesting. I did not do that myself. I just used F2 (rename) on `n`